File: Programming Basics/C++/While loops/Walking/Walking.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
	const int goalLimit = 10000;

	int totalSteps = 0;
	bool shouldContinue = true, goalIsReached = false;

	while (shouldContinue && !goalIsReached) {
		string input;
		getline(cin, input);

		int currentSteps;
		if (input == "Going home") {
			cin >> currentSteps;
			shouldContinue = false;
		}
		else currentSteps = stoi(input);

		totalSteps += currentSteps;
		if (totalSteps >= goalLimit) goalIsReached = true;
	}

	if (goalIsReached) cout << "Goal reached! Good job!" << endl;
	else cout << goalLimit - totalSteps << " more steps to reach goal." << endl;
}

File: Programming Basics/JavaScript/While loops/Exercises/walking.js
function solve(input) {
    const goalLimit = 10000;

    let totalSteps = 0;
    let shouldContinue = true;
    let goalIsReached = false;
    let index = 0;

    while (shouldContinue && !goalIsReached) {
        let stepsInput = input[index];

        if (stepsInput === "Going home") {
            index++;
            shouldContinue = false;
        }

        let currentSteps = Number(input[index]);

        totalSteps += currentSteps;
        if (totalSteps >= goalLimit) goalIsReached = true;

        index++;
    }

    if (goalIsReached) {
        console.log("Goal reached! Good job!");
        console.log(`${totalSteps - goalLimit} steps over the goal!`);
    } else console.log(`${goalLimit - totalSteps} more steps to reach goal.`);
}
